Bootloader: Allow redefining of DFU wait period

kubark42 opened the pull request:
Previously, it was always 6 seconds. In practice, this doesn't seem always necessary. This allows redefining the delay for individual targets.

We find that a 2 second wait isn't causing us any issues and is a lot more tolerable than 6 seconds when debugging.

peabody124 commented on flight/targets/bl/common/main.c:
```diff
@@ -49,7 +49,11 @@ extern void PIOS_Board_Init(void);
 #endif
 
+#if defined(DFU_PAUSE_DELAY_MS) // allow altering bootloader delay
+#define BL_WAIT_FOR_DFU_TIMER MSEC_TO_USEC(DFU_PAUSE_DELAY_MS)
```
Just to write this more typically could you do this?

```
#ifndef DFU_PAUSE_DELAY_MS
#define DFU_PAUSE_DELAY_MS 6
#endif

#define BL_WAIT_FOR_DFU_TIMER MSEC_TO_USEC(DFU_PAUSE_DELAY_MS)
```

Less lines

kubark42 replied:
Yup, done.

mlyle commented:
Some old netbooks need nearly 4 seconds to enumerate and catch it, so we decided to keep it at 6 seconds in dR-- the approach we took is we have GCS kick it out of the loader / trigger boot.  If it enumerates quicker than 2 seconds, it boots even faster; it's still correct in the other case.

kubark42 committed:
Bootloader: Allow redefining of DFU wait period

Previously, it was always 6 seconds. In practice, this doesn't seem always
necessary. This allows redefining the delay for individual targets.
